ludwig/data/preprocessing.py

```diff
@@ -1101,6 +1102,16 @@ def build_dataset(
     for callback in callbacks or []:
         callback.on_build_metadata_end(dataset_df, mode)
 
+    # Happens after metadata is built so we can use precomputed fill values.
+    logger.debug("handle missing values")
+    for feature_config in feature_configs:
+        preprocessing_parameters = metadata[feature_config[NAME]][PREPROCESSING]
+        handle_missing_values(dataset_cols, feature_config, preprocessing_parameters)
```

Comment:
I just noticed that `handle_missing_values` is also called in `build_metadata()`, so we'll be making the call to handling missing values twice. Is that expected?

Reply:
I may be misremembering, but I believe @tgaddair originally introduced the double missing values call for a reason

Reply:
Just pushed up a refactor that makes the call to `handle_missing_values` once. I couldn't find a good reason for calling it twice, but super happy to revert this change @tgaddair if I missed something.

Reply:
Just pushed up a refactor that makes the call to `handle_missing_values` once. I couldn't find a good reason for calling it twice, but super happy to revert this change @tgaddair if I missed something.

Reply:
the hdf5 is supposed to be output of internally processing, you cannot throw a random unprocessed hdf5 at it